core/src/test/java/org/infinispan/expiration/impl/ExpirationFunctionalTest.java

```diff
@@ -49,7 +49,7 @@ public void testSimpleExpirationLifespan() throws Exception {
    public void testSimpleExpirationMaxIdle() throws Exception {
 
       for (int i = 0; i < SIZE; i++) {
-         cache.put("key-" + i, "value-" + i,-1, null, 1, TimeUnit.MILLISECONDS);
+         cache.put("key-" + i, "value-" + i,-1, TimeUnit.MILLISECONDS, 1, TimeUnit.MILLISECONDS);
```

Comment:
I believe the problem should be addressed in the Metadata implementation. Negative time values should not be converted, so the given TimeUnit could be null and not result in an exception.